Neos.Flow/Tests/Functional/Mvc/ActionControllerTest.php

```diff
@@ -520,4 +530,34 @@ public function trustedPropertiesConfigurationDoesNotIgnoreWildcardConfiguration
         $response = $this->browser->sendRequest($request);
         self::assertSame('Entity "Foo" updated', $response->getBody()->getContents());
     }
+
+    /**
+     * @test
+     */
+    public function flashMessagesGetRenderedAfterRedirect()
+    {
+        $request = $this->serverRequestFactory->createServerRequest('GET', new Uri('http://localhost/test/mvc/actioncontrollertest/redirectWithFlashMessage'));
+        $response = $this->browser->sendRequest($request);
+
+        $sessionCookies = array_map(static function ($cookie) {
+            return Cookie::createFromRawSetCookieHeader($cookie);
+        }, $response->getHeader('Set-Cookie'));
+        self::assertNotEmpty($sessionCookies);
+
+        $redirect = $response->getHeaderLine('Location');
+        self::assertNotEmpty($redirect);
+
+        $this->objectManager->forgetInstance(StandardController::class);
```

Comment:
Not sure this is really needed, but through the `AbstractController` it holds a reference to `ControllerContext` which holds a reference to `FlashMessageService`, so...